### PR TITLE
Proxy flags

### PR DESCRIPTION
**Pull request by yulianedyalkova**
Pass proxy values from flags

### CODE REVIEW

**Comment by tfussell** [comment]
Towards https://github.com/giantswarm/giantswarm/issues/17252

File: main.go [hunk 1]
```diff
@@ -128,6 +128,9 @@ func mainError() error {
 	daemonCommand.PersistentFlags().StringSlice(f.Service.Registry.Mirrors, []string{}, `Image registry mirror domains. Can be set only if registry domain is "docker.io".`)
 
 	daemonCommand.PersistentFlags().String(f.Service.Tenant.Ignition.Path, "/opt/ignition", "Default path for the ignition base directory.")
+	daemonCommand.PersistentFlags().String(f.Service.Tenant.Proxy.HTTP, "", "Where to proxy HTTP trafic to.")
```
Comment (tfussell) [inline main.go:131]:
```suggestion
	daemonCommand.PersistentFlags().String(f.Service.Tenant.Proxy.HTTP, "", "URL of proxy for HTTP requests.")
```

File: main.go [hunk 2]
```diff
@@ -128,6 +128,9 @@ func mainError() error {
...
+	daemonCommand.PersistentFlags().String(f.Service.Tenant.Proxy.HTTPS, "", "Where to proxy HTTPS trafic to.")
```
Comment (tfussell) [inline main.go:132]:
```suggestion
	daemonCommand.PersistentFlags().String(f.Service.Tenant.Proxy.HTTPS, "", "URL of proxy for HTTPS requests.")
```

File: service/controller/cloudconfig/cloud_config.go [hunk 3]
```diff
@@ -85,10 +87,17 @@ type CloudConfig struct {
 	dockerhubToken  string
 	ignitionPath    string
 	k8sAPIExtraArgs []string
+	proxy           proxyConfig
 	registryMirrors []string
 	ssoPublicKey    string
 }
 
+type proxyConfig struct {
```
Comment (tfussell) [inline service/controller/cloudconfig/cloud_config.go:95]: Maybe we could drop this and just use `ProxyConfig` (the exported one) in the `CloudConfig` struct?

Comment (yulianedyalkova) [reply service/controller/cloudconfig/cloud_config.go:95]: The exported one has `[]string` type for `NoProxy` where the private one already has the values joined so it's just a string.

Comment (tfussell) [reply service/controller/cloudconfig/cloud_config.go:95]: They could also be joined here https://github.com/giantswarm/kvm-operator/blob/02dc26a6dafc069275ac11a1f8303d9f800675af/service/controller/cloudconfig/master_template.go#L52 or in the template itself, but it's fine if you prefer this way.

**Review by tfussell — APPROVED** [review]
LGTM.

File: service/controller/cloudconfig/cloud_config.go (hunk 3 above)
Comment (yulianedyalkova) [reply service/controller/cloudconfig/cloud_config.go:95]: I will leave it as it is for now cause I don't have time to retest now and we need it asap. I can clean it up in a future commit if that's ok with you :)

**Pull request merged**